askama_shared/src/parser.rs

```diff
@@ -51,6 +51,44 @@ pub enum Expr<'a> {
     RustMacro(&'a str, &'a str),
 }
 
+impl Expr<'_> {
+    /// Returns `true` if enough assumptions can be made,
+    /// to determine that `self` is copyable.
+    pub fn is_copyable(&self) -> bool {
+        fn is_copyable(expr: &Expr, within_op: bool) -> bool {
+            use Expr::*;
+            match expr {
+                BoolLit(_) | NumLit(_) | StrLit(_) | CharLit(_) => true,
+                Unary(.., expr) => is_copyable(expr, true),
+                BinOp(_, lhs, rhs) => is_copyable(lhs, true) && is_copyable(rhs, true),
+                // FIXME: The parser should produce `NumLit(_)` instead of `Attr(NumLit(_), _)` for floats
```

Comment:
Should I just resolve this in this PR? As in a prior commit.

Reply:
Yeah, that would be great!